### ISSUE TITLE
Out of bounds array access during C-Fortran string conversion

### ISSUE THREAD

**Issue by IanH0073** [vmagnin/gtk-fortran#82]
In numerous places in the examples and main source there are a number of constructs similar to:

```
call C_F_POINTER(some_cptr_returning_function(xyx), fortran_char_array, (/0/))
!... operations on fortran_char_array(1:), some of which are to detect the 
! null character in the C string
```

This pattern is non-conforming fortran source.  A series of patches to fix to follow.

```
Find all "(/0/)", Subfolders, Keep modified files open, Find Results 1, "Entire Solution", "*.f90;*.for;*.f;*.fi;*.fd"
  H:\Projects\gtk-fortran\examples\bazaar.f90(128):    call C_F_POINTER(gdk_pixbuf_get_pixels(my_pixbuf), pixel, (/0/))
  H:\Projects\gtk-fortran\examples\bazaar.f90(182):    call C_F_POINTER(gtk_entry_get_text(entry1), textptr, (/0/))
  H:\Projects\gtk-fortran\examples\bazaar.f90(249):    call C_F_POINTER(gtk_file_chooser_get_filename (widget), textptr, (/0/))
  H:\Projects\gtk-fortran\examples\cairo-tests.f90(161):  call c_f_pointer(gdk_pixbuf_get_pixels(my_pixbuf), pixel, (/0/))
  H:\Projects\gtk-fortran\src\gtk-sup.f90(664):    call c_f_pointer(ctext, textptr, (/0/))
  H:\Projects\gtk-fortran\src\gtk-sup.f90(664):    call c_f_pointer(ctext, textptr, (/0/))
  H:\Projects\gtk-fortran\examples\julia_pixbuf.f90(156):    call C_F_POINTER(gtk_combo_box_text_get_active_text(combo1), textptr, (/0/))
  H:\Projects\gtk-fortran\examples\julia_pixbuf.f90(392):  call c_f_pointer(gdk_pixbuf_get_pixels(my_pixbuf), pixel, (/0/))
  H:\Projects\gtk-fortran\examples\mandelbrot_pixbuf.f90(115):  call c_f_pointer(gdk_pixbuf_get_pixels(my_pixbuf), pixel, (/0/))
  H:\Projects\gtk-fortran\examples\menu.f90(263):    call c_f_pointer(error, textptr, (/0/))
  Matching lines: 10    Matching files: 7    Total files searched: 90
```

**Comment by vmagnin** [comment]
The Fortran 2018 draft says about  C_F_POINTER (CPTR, FPTR [, SHAPE]):
> SHAPE (optional) shall be a rank-one integer array. It is an INTENT (IN) argument. SHAPE shall be present  if and only if FPTR is an array; its size shall be equal to the rank of FPTR.

The array (/0/) is a rank-one integer array containing one element, so its size is 1. And we need Fortran pointers toward a rank 1 array.

**Comment by vmagnin** [comment]
I reopen this issue.

- I have just corrected the four `call c_f_pointer(gdk_pixbuf_get_pixels(my_pixbuf), pixel, (/0/))`: see issue https://github.com/vmagnin/gtk-fortran/issues/130
- And I am now working on the `call c_f_pointer(ctext, textptr, (/0/))` problems: issue https://github.com/vmagnin/gtk-fortran/issues/133

**Comment by vmagnin** [comment]
I have applied your C F string fixes (see issue #133) in master(gtk2) and gtk3 branches.
